### PR TITLE
Notification Bar

### PR DESCRIPTION
**Pull request by Anntol**
Show error or success messages

### CODE REVIEW

File: rslang/src/app/auth/services/auth-action.service.ts [hunk 1]
```diff
@@ -15,26 +16,48 @@ export class AuthActionService extends BaseActionService {
...
+        notifyService.showSuccess('You are logged in!');
+      },
+      onError(error) {
+        let message = '';
+        switch (error.status) {
+          case 403:
+            message = 'Password is incorrect!';
+            break;
+          case 404:
+            message = 'User is not found!';
```
Comment (Nikitzu) [inline rslang/src/app/auth/services/auth-action.service.ts:55]: maybe Page?

File: rslang/src/app/shared/components/notification/notification-bar.component.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+@Component({
+  selector: 'app-notification',
+  templateUrl: './notification-bar.component.html',
+})
+export class NotificationBar {
+  constructor(public snackBar: MatSnackBar) {}
+
+  openSnackBar(message: string, className: string): void {
+    this.snackBar.open(message, undefined, {
```
Comment (Nikitzu) [inline rslang/src/app/shared/components/notification/notification-bar.component.ts]: undefined as a parameter??

**Review by Nikitzu — CHANGES_REQUESTED** [review]

File: rslang/src/app/auth/services/auth-action.service.ts (hunk 1 above)
Comment (Anntol) [reply rslang/src/app/auth/services/auth-action.service.ts:55]: 404 is returned from the backend when email is not found in db (it is user signing process)

File: rslang/src/app/shared/components/notification/notification-bar.component.ts (hunk 2 above)
Comment (Anntol) [reply rslang/src/app/shared/components/notification/notification-bar.component.ts]: fixed

File: rslang/src/app/auth/services/auth-action.service.ts [hunk 3]
```diff
@@ -15,26 +16,48 @@ export class AuthActionService extends BaseActionService {
...
+          : 'Sign up error! Please try again.';
+        notifyService.showError(message);
       },
-      // TODO catch error
     };
     this.sendAction('POST', `${BASE_URL}/users`, action, { body: req });
   }
 
-  signinUser(req: SigninRequest, authService = this.authService): void {
+  signinUser(req: SigninRequest, authService = this.authService, notifyService = this.notifyService): void {
```
Comment (Nikitzu) [inline rslang/src/app/auth/services/auth-action.service.ts]: Why we have services as a parameter of the method????

**Review by Nikitzu — CHANGES_REQUESTED** [review]

File: rslang/src/app/auth/services/auth-action.service.ts (hunk 3 above)
Comment (Anntol) [reply rslang/src/app/auth/services/auth-action.service.ts]: fixed

**Review by Nikitzu — APPROVED** [review]

**Review by kastrubait — APPROVED** [review]

**Pull request merged**